internal/encoding/block/from_parquet.go

```diff
@@ -21,17 +21,14 @@ func FromParquetBy(payload []byte, partitionBy string, filter *typeof.Schema, ap
 	// Find the partition index
 	schema := iter.Schema()
 	cols := schema.Columns()
-	partitionIdx, ok := findString(cols, partitionBy)
-	if !ok {
-		return nil, nil // Skip the file if it has no partition column
-	}
 
 	// The resulting set of blocks, repartitioned and chunked
 	blocks := make([]Block, 0, 128)
 
 	// Create presto columns and iterate
 	result, size := make(map[string]column.Columns, 16), 0
-	_, _ = iter.Range(func(rowIdx int, r []interface{}) bool {
+
```

Comment:
I don't think the implementation is correct `filter *typeof.Schema, apply applyFunc` should work since it's required by the contract implementation of this. This completely ignores the filter schema and transformations. See `orc` as a reference implementation of how it should work: https://github.com/kelindar/talaria/blob/master/internal/encoding/block/from_orc.go#L20